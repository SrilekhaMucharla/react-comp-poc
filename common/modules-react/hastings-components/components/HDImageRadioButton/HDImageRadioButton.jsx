import PropTypes from 'prop-types';
import React from 'react';
import classnames from 'classnames';
import onlineProductLogo from '../../../../../applications/quote-and-buy/src/assets/images/logo/logo-online-product.svg';
import standardProductLogo from '../../../../../applications/quote-and-buy/src/assets/images/logo/logo-standard-product.svg';


const HDImageRadioButton = ({
    className, path, name, currentValue, value, onChange, onBlur, image, labelValue, secondaryLabel, selected, hideInput, type, onlineProduct
}) => {
    return (
        <div className={`hd-image-radio-btn${(className) ? ` ${className}` : ''}${(selected) ? ' selected' : ''}`}>
            {image ? (
                <div>
                    <img
                        className={classnames(`hd-image-radio-btn__image ${onlineProduct ? 'online-product-on' : ''}`, { selected })}
                        src={image}
                        alt={currentValue} />
                    {onlineProduct && (
                        <img
                            className={classnames('hd-image-radio-btn__image_online', { selected })}
                            src={currentValue === 'YD' ? standardProductLogo : onlineProductLogo}
                            alt="Online product logo" />
                    )}
                    {secondaryLabel && <div className="hd-image-radio-btn__inverted-text">{secondaryLabel}</div>}
                </div>
            ) : (
                <div className="hd-image-radio-btn__header-value">
                    {currentValue}
                    {secondaryLabel && <div className="hd-image-radio-btn__inverted-text">{secondaryLabel}</div>}
                </div>
            )}
            {labelValue ? <div className="label-text">{labelValue}</div> : ''}
            {!hideInput && (
                <input
                    className="hd-image-radio-btn_input"
                    data-testid="radio-btn"
                    type={type}
                    path={path}
                    name={name}
                    value={currentValue}
                    checked={currentValue === value}
                    onChange={onChange}
                    onBlur={onBlur} />
            )}
        </div>
    );
};

HDImageRadioButton.propTypes = {
    className: PropTypes.string,
    path: PropTypes.string,
    name: PropTypes.string,
    currentValue: PropTypes.string.isRequired,
    value: PropTypes.string,
    onChange: PropTypes.func,
    onBlur: PropTypes.func,
    image: PropTypes.node,
    labelValue: PropTypes.string,
    secondaryLabel: PropTypes.string,
    selected: PropTypes.bool,
    hideInput: PropTypes.bool,
    onlineProduct: PropTypes.bool,
    type: PropTypes.string
};

HDImageRadioButton.defaultProps = {
    className: '',
    path: 'path.to.radio-button',
    name: 'radio-button-name',
    value: '',
    onChange: () => { },
    onBlur: () => { },
    labelValue: '',
    image: null,
    secondaryLabel: null,
    selected: false,
    hideInput: false,
    onlineProduct: false,
    type: 'radio'
};

// avoid obfuscation problem with types
HDImageRadioButton.typeName = 'HDImageRadioButton';

export default HDImageRadioButton;
