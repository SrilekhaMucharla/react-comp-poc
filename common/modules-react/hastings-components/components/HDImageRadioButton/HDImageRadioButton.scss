.hd-image-radio-btn {
    display: flex;
    flex-direction: column;
    align-items: center;
    height: 100%;
}

.hd-image-radio-btn__header-value {
    font-style: normal;
    font-weight: bold;
    font-size: 14px;
    line-height: 120%;
    text-align: center;
    flex-grow: 1;
    padding: 20px 0 10px 0;

    @include media-breakpoint-up(md) {
        font-size: 16px;
    }
}

.hd-image-radio-btn__inverted-text {
    margin: 15px 0 0 0;
    padding: 6px 6px 6px;
    border-radius: 2px;
    background-color: $app-theme;
    font-size: 12px;
    font-weight: bold;
    font-stretch: normal;
    font-style: normal;
    line-height: 1;
    letter-spacing: normal;
    text-align: center;
    color: $mono-white;
    position: relative;

    &:after {
        content: "";
        position: absolute;
        bottom: 100%;
        left: 50%;
        margin-left: -10px;
        width: 0;
        height: 0;
        border-left: 10px solid transparent;
        border-right: 10px solid transparent;
        border-bottom: 10px solid $app-theme;
    }
}

.hd-image-radio-btn__image {
    height: 35px;
    width: 100%;
    margin-top: 37px;
    margin-bottom: 29px;

    &.selected {
        filter: brightness(0) invert(1);
    }

    &.online-product-on {
        margin-bottom: 5px;
    }
}

.hd-image-radio-btn__image_online {
    height: 35px;
    width: auto;
    margin-top: 5px;
    margin-bottom: 15px;
    display: flex;

    &.selected {
        filter: brightness(0) invert(1);
    }
}

.hd-image-radio-btn_input {
    cursor: pointer;
    height: 30px;
    width: 30px;
    position: relative;
    margin: 10px 0;
    border: 2.5px solid $info-color;
    border-radius: 50%;
    display: flex;
    align-items: center;
    justify-content: center;
    -webkit-appearance: none !important;
    -moz-appearance: none !important;
    appearance: none !important;

    &:checked {
        border: 2.5px solid $mono-white;
    }

    &:checked:after {
        content: "";
        height: 22px;
        width: 22px;
        background-color: $mono-white;
        border: 3px solid $midnight-royal-blue;
        border-radius: 50%;
    }
}
