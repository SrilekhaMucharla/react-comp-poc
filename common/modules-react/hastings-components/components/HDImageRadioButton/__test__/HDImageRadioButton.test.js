import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import HDImageRadioButton from '../HDImageRadioButton';

Enzyme.configure({ adapter: new Adapter() });

describe('<HDImageRadioButton />', () => {
    it('render component with image', () => {
        const onChangeMock = jest.fn();
        const component = (
            <HDImageRadioButton
                name="group"
                currentValue="variant1"
                value="variant1"
                onChange={onChangeMock}
                image={<img alt="HD" />} />
        );
        const wrapper = shallow(component);
        expect(wrapper).toMatchSnapshot();
    });

    it('Change value and check after select radio button', () => {
        const onChangeMock = jest.fn();
        const event = {
            preventDefault() { },
            target: { value: 'variant2' }
        };
        const component = (
            <HDImageRadioButton
                name="group"
                currentValue="variant2"
                value="variant1"
                onChange={onChangeMock}
                image={<img alt="HD" />} />
        );
        const wrapper = shallow(component);
        const radioButton = wrapper.find({ 'data-testid': 'radio-btn' });
        radioButton.simulate('change', event);
        setTimeout(() => {
            expect(radioButton.prop('value')).toBe('variant2');
            expect(radioButton.prop('checked')).toBe(true);
        }, 0);
    });
});
