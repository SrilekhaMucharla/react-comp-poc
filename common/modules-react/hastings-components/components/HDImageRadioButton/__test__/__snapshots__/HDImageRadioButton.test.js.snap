// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HDImageRadioButton /> render component with image 1`] = `ShallowWrapper {}`;
